failure to compile with OpenMPI 4.0 due to removed symbols

```rust
error[E0425]: cannot find function `MPI_Type_hvector` in module `ffi` 
  --> /home/cc/.cargo/registry/src/github.com-1ecc6299db9ec823/mpi-0.5.4/src/datatype.rs:212:18 
   | 
212 |             ffi::MPI_Type_hvector(count, blocklength, stride, oldtype.as_raw(), &mut newtype); 
   |                  ^^^^^^^^^^^^^^^^ did you mean `MPI_Type_vector`? 

error: aborting due to previous error
```

OpenMPI 4.0 (which recently landed on arch) removes some symbols that were eliminated from the MPI standard 3.0.

https://www.open-mpi.org/faq/?category=mpi-removed

--- a/src/datatype.rs
+++ b/src/datatype.rs
@@ -473,7 +473,13 @@ impl UncommittedUserDatatype {
     {
         let mut newtype: MPI_Datatype = unsafe { mem::uninitialized() };
         unsafe {
-            ffi::MPI_Type_hvector(count, blocklength, stride, oldtype.as_raw(), &mut newtype);
+            ffi::MPI_Type_create_hvector(
+                count,
+                blocklength,
+                stride,
+                oldtype.as_raw(),
+                &mut newtype,
+            );
         }
         UncommittedUserDatatype(newtype)
     }

